python/cudf/cudf/core/index.py

```diff
@@ -2130,6 +2221,18 @@ def str(self):
     def _constructor_expanddim(self):
         return cudf.MultiIndex
 
+    def _clean_nulls_from_index(self):
+        """
+        Convert all na values(if any) in Index object
+        to `None` as a preprocessing step to `__repr__` methods.
+        """
+        if self._values.has_nulls:
+            return cudf.Index(
+                self._values.astype("str").fillna("None"), name=self.name
```

Comment:
Should this should be `<NA>` instead of `None`?

Reply:
Changed this to `<NA>` uniformly everywhere - So now even incase of string dtypes we will have `<NA>` instead of `None`